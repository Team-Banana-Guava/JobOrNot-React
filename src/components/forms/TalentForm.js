import React, { Component, PropTypes } from 'react';
import SingleInput from './SingleInput';
import Nav from '../navbar/Nav';
import { connect } from 'react-redux';
import { sendSignUp } from '../../actions/auth-actions';

class TalentForm extends Component {
    constructor(props) {
        super(props);
        this.state = {
            firstName: '',
            lastName: '',
            username: '',
            email: '',
            password: ''
        };

        this.handleFormSignUp = this.handleFormSignUp.bind(this);
        // this.handleFormUpdate = this.handleFormUpdate.bind(this);
        this.handleFormClear = this.handleFormClear.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleFormSignUp(e) {
        e.preventDefault();

        const formPayload = {
            firstName: this.state.firstName,
            lastName: this.state.lastName,
            username: this.state.username,
            email: this.state.email,
            password: this.state.password,
            role: 'talent'
        };

        this.props.signUp({ method: 'POST', path: '/signup', body: formPayload });
        
        this.handleFormClear(e);
    }

    // handleFormUpdate() {
    //     // update form logic goes here
    // }

    handleFormClear(e) {
        e.preventDefault();
        this.setState({
            firstName: '',
            lastName: '',
            username: '',
            email: '',
            password: '',
        });
    }

    handleChange(e) {
        this.setState({ [e.target.name]: e.target.value });
    }
 
    render() {
        return (
            <div>
                <Nav />
                <form onSubmit={this.handleFormSignUp}>
                    <h1>THIS FORM SIGNS UP TALENT</h1>
                    <SingleInput 
                        title={'First Name'}
                        name={'firstName'}
                        inputType={'text'}
                        content={this.state.firstName}
                        controlFunc={this.handleChange}
                        placeholder={'First Name'} />
                    <SingleInput 
                        title={'Last Name'}
                        name={'lastName'}
                        inputType={'text'}
                        content={this.state.lastName}
                        controlFunc={this.handleChange}
                        placeholder={'Last Name'} />
                    <SingleInput 
                        title={'User Name'}
                        name={'username'}
                        inputType={'text'}
                        content={this.state.username}
                        controlFunc={this.handleChange}
                        placeholder={'Select a User Name'} />
                    <SingleInput 
                        title={'Email Address'}
                        name={'email'}
                        inputType={'text'}
                        content={this.state.email}
                        controlFunc={this.handleChange}
                        placeholder={'Email Address'} />
                    <SingleInput 
                        title={'Password'}
                        name={'password'}
                        inputType={'password'}
                        content={this.state.password}
                        controlFunc={this.handleChange}
                        placeholder={'Select a Password'} />
                    <button onClick={this.handleFormClear}>
                        Clear Form
                    </button>
                    <input 
                        type='submit'
                        value='submit'/>
                </form>
            </div>
        );
    };
}

function mapStateToProps(state) {
    return {
        resume: state.uploads
    };
}

function mapDispatchToProps(dispatch) {
    return {
        signUp: (options) => dispatch(sendSignUp(options))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(TalentForm);

TalentForm.propTypes = {
    signUp: PropTypes.func,
    resume: PropTypes.object
};
