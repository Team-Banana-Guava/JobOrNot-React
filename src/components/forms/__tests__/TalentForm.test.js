/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TalentForm from '../TalentForm';

describe('TalentForm', function () {
    it('renders the TalentForm form/component', () => {
        const rendered = shallow(<TalentForm signUp={() => {}} signedIn={true}/>);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});
