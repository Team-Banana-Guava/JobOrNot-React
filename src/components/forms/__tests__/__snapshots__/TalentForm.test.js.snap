exports[`TalentForm renders the TalentForm form/component 1`] = `
<div>
  <Nav />
  <form
    onSubmit={[Function]}>
    <h1>
      THIS FORM SIGNS UP TALENT
    </h1>
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="firstName"
      placeholder="First Name"
      title="First Name" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="lastName"
      placeholder="Last Name"
      title="Last Name" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="userName"
      placeholder="Select a User Name"
      title="User Name" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="email"
      placeholder="Email Address"
      title="Email Address" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="password"
      name="password"
      placeholder="Select a Password"
      title="Password" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="resume"
      placeholder="Holding box for resume uploads"
      title="Resume" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="skills"
      placeholder="Top Skills"
      title="Skills" />
    <SingleInput
      content=""
      controlFunc={[Function]}
      inputType="text"
      name="locations"
      placeholder="Locations"
      title="Locations" />
    <button
      onClick={[Function]}>
      Clear Form
    </button>
    <input
      type="submit"
      value="submit" />
  </form>
</div>
`;
